util/chunk/chunk.go

```diff
@@ -30,24 +32,60 @@ type Chunk struct {
 	// numVirtualRows indicates the number of virtual rows, which have zero column.
 	// It is used only when this Chunk doesn't hold any data, i.e. "len(columns)==0".
 	numVirtualRows int
+	// capacity indicates the max number of rows that the chunk wants to hold.
+	// The accurate row count after executor.Next is guaranteed by executor implement.
+	// It is used only during executor's execution, and grow by `Renew` or `Reset`.
+	capacity int
 }
 
 // Capacity constants.
 const (
 	InitialCapacity = 32
+	varElemLen      = -1
 )
 
 // NewChunkWithCapacity creates a new chunk with field types and capacity.
 func NewChunkWithCapacity(fields []*types.FieldType, cap int) *Chunk {
+	return New(fields, cap, cap) //FIX ME in continue PR, this keep every thing same.
```

Comment:
In the next PR, we remove this function and use `New` instead?

Reply:
Yes~